src/main/java/javaslang/collection/Iterator.java

```diff
@@ -128,6 +128,34 @@ public T next() {
     }
 
     /**
+     * Creates an Iterator which traverses along all given iterators.
+     * @param iterators The list of iterators
+     * @param <T>       Component type.
+     * @return A new {@code javaslang.collection.Iterator}
+     */
+    @SafeVarargs
+    @SuppressWarnings("unchecked")
+    static <T> Iterator<T> ofIterators(Iterator<? extends T>... iterators) {
+        Objects.requireNonNull(iterators, "iterators is null");
+        List<Iterator<T>> queue = List.empty();
+        for (int i = iterators.length - 1; i >= 0; i--) {
+            queue = queue.prepend((Iterator<T>) iterators[i]);
+        }
+        return queue.isEmpty() ? Iterator.empty() : new ConcatIterator<>(queue);
+    }
+
+    /**
+     * Creates an Iterator which traverses along all given iterators.
+     * @param iterators The list of iterators
+     * @param <T>       Component type.
+     * @return A new {@code javaslang.collection.Iterator}
+     */
+    static <T> Iterator<T> ofIterators(Traversable<Iterator<T>> iterators) {
```

Comment:
thank you, but I'll use <code>Iterator&lt;Iterator&lt;T&gt;&gt;</code>
see [this](#issuecomment-130607691) comment